addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/handler/SenseBoxHandler.java

```diff
@@ -0,0 +1,374 @@
+            return false;
+        }
+
+        return true;
+    }
+
+    private final ExpiringCache<String, SenseBoxData> CACHE = new ExpiringCache<>(CACHE_EXPIRY,
+            new ExpiringCache.LoadAction<String, SenseBoxData>() {
+                @Override
+                public SenseBoxData load(String senseBoxId) throws IOException {
+                    SenseBoxData result = reallyFetchDataFromServer(senseBoxId);
+                    return result;
+                }
+            });
+
+    private SenseBoxData reallyFetchDataFromServer(String senseBoxId) {
+        String query = SENSEMAP_API_URL_BASE + "/boxes/" + senseBoxId;
+
+        // the caching layer does not like null values
+        SenseBoxData result = new SenseBoxData();
+
+        try {
+            URL url = new URL(query);
+            URLConnection connection = url.openConnection();
+            String body = IOUtils.toString(connection.getInputStream());
+
+            logger.debug("Fetched Data");
+            logger.debug(body);
+            SenseBoxData parsedData = gson.fromJson(body, SenseBoxData.class);
+
+            // Could perhaps be simplified via triply-nested arrays
+            // http://stackoverflow.com/questions/36946875/how-can-i-parse-geojson-with-gson
+            for (SenseBoxLoc loc : parsedData.getLocs()) {
+                if (loc.getGeometry() != null) {
+                    List<Double> locationData = loc.getGeometry().getData();
+                    if (locationData != null) {
+                        SenseBoxLocation location = new SenseBoxLocation();
+
+                        if (locationData.size() > 0) {
+                            location.setLongitude(locationData.get(0));
+                        }
+
+                        if (locationData.size() > 1) {
+                            location.setLatitude(locationData.get(1));
+                        }
+
+                        if (locationData.size() > 2) {
+                            location.setHeight(locationData.get(2));
+                        }
+
+                        parsedData.setLocation(location);
+                    }
+                }
+            }
+
+            for (SenseBoxSensor sensor : parsedData.getSensors()) {
+                if ("VEML6070".equals(sensor.getSensorType())) {
+                    // "unit" is not nicely comparable, so use sensor type for now
+                    parsedData.setUvIntensity(sensor);
+                } else if ("lx".equals(sensor.getUnit())) {
+                    parsedData.setLuminance(sensor);
+                } else if ("hPa".equals(sensor.getUnit())) {
+                    parsedData.setPressure(sensor);
+                } else if ("%".equals(sensor.getUnit())) {
+                    parsedData.setHumidity(sensor);
+                } else if ("°C".equals(sensor.getUnit())) {
+                    parsedData.setTemperature(sensor);
+                } else {
+                    logger.debug("    Sensor: {}", sensor);
```

Comment:
May be, but in that case, it helps immensely tracing API connection problems without creating costly toString() methods on the value holder objects. I would prefer to keep this as it is.